packages/ai/src/ai/common/validation.py

```
@@ -0,0 +1,135 @@
+    Args:
+        output_tokens: The configured max output tokens.
+        total_tokens: The model's total context window.
+
+    Returns:
+        The validated output token value (clamped if necessary).
+
+    Raises:
+        ValueError: If output_tokens is not a positive integer.
+    """
+    if not isinstance(output_tokens, int) or output_tokens < 1:
+        raise ValueError(f'Output tokens must be a positive integer, got {output_tokens!r}.')
```

Ruff (0.15.7)

[warning] Avoid specifying long messages outside the exception class

(TRY003)